runners/spark/spark_runner.gradle

```diff
@@ -165,12 +165,12 @@ dependencies {
   implementation library.java.vendored_grpc_1_43_2
   implementation library.java.vendored_guava_26_0_jre
   implementation "com.codahale.metrics:metrics-core:3.0.1"
-  implementation "org.apache.spark:spark-core_$spark_scala_version:$spark_version"
-  implementation "org.apache.spark:spark-network-common_$spark_scala_version:$spark_version"
+  provided "org.apache.spark:spark-core_$spark_scala_version:$spark_version"
+  provided "org.apache.spark:spark-network-common_$spark_scala_version:$spark_version"
   permitUnusedDeclared "org.apache.spark:spark-network-common_$spark_scala_version:$spark_version"
```

Comment:
I don't think we need `permitUnusedDeclared` for provided dependencies.

Reply:
I ran into some issues with this with other provided deps :/